File: Frontend/src/app/models/web-socket-message.ts
export enum MsgType {
    Connection = 'Connection',
    FriendListUpdate = 'FriendListUpdate',
    FriendStatusUpdate = 'FriendStatusUpdate',
    FriendRequestUpdate = 'FriendRequestUpdate',
    GameRoom = 'GameRoom',
    StartGame = 'StartGame',
    GameInvitation = 'GameInvitation',
    CancelGameInvitation = 'CancelGameInvitation',
    StatsUpdate = 'StatsUpdate',
    GameStart = 'GameStart',
    GameUpdate = 'GameUpdate',
    GameOver = 'GameOver',
    Chat = 'Chat',
    RematchRequest = 'RematchRequest',
    CancelRematchRequest = 'CancelRematchRequest',
    UserBanned = 'UserBanned'
}

export interface WebSocketMessage {
    Type: MsgType;
    Id: number;
    Content: any;
}

File: Backend/JuegoWeb/Models/Dtos/WebSocketMessage.cs
using System.Text.Json.Serialization;

namespace JuegoWeb.Models.Dtos;

public enum MsgType
{
    Connection,             // Conexión con el WebSocket
    FriendListUpdate,       // Actualizar la lista de amigos
    FriendStatusUpdate,     // Actualizar estados de los amigos
    FriendRequestUpdate,    // Notificar de solicitudes de amistad
    GameRoom,               // Gestionar una sala de juego
    StartGame,              // Notificar del inicio de la partida al usuario invitado
    GameInvitation,         // Invitación de un amigo
    CancelGameInvitation,   // Cancelar la invitación
    StatsUpdate,            // Estadísticas globales
    GameStart,              // Comenzarla partida
    GameUpdate,             // Actualizaciones de la partida
    GameOver,               // Finalizar partida
    Chat,                   // Mensaje del chat
    RematchRequest,         // Solicitud de revancha
    CancelRematchRequest    // Cancelar revancha
}

public class WebSocketMessage
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public MsgType Type { get; set; }

    public int Id { get; set; }

    public object Content { get; set; }
}